lib/event-bus/src/rabbit-event-bus/amqp-connector.test.ts

```diff
@@ -0,0 +1,239 @@
+            expect(mockChannel.assertQueue).toHaveBeenCalledWith('consumer__foo-bar__service');
+            expect(mockChannel.bindQueue).toHaveBeenCalledTimes(1);
+            expect(mockChannel.bindQueue).toHaveBeenCalledWith('consumer__foo-bar__service', 'event__foo-bar', '');
+            expect(mockChannel.consume).toHaveBeenCalledTimes(1);
+            expect(mockChannel.consume.mock.calls[0][0]).toBe('consumer__foo-bar__service');
+        });
+
+        it('it should call the subscription handler and acknowledge', async () => {
+            const mockChannel = makeChannel({
+                assertQueue: jest.fn().mockImplementation(() => Promise.resolve()),
+                consume: (___, callback) => {
+                    callback({
+                        content: { toString: () => '{ "event": "foo" }' },
+                    });
+                },
+            });
+            const mockConnection = makeConnection(mockChannel);
+
+            // tslint:disable-next-line: no-any
+            (connect as any).mockImplementation(async (): Promise<Connection> => mockConnection);
+            const connector = new AMQPConnector<{}>(url, channel(), [], [], 'service');
+            const handler = jest.fn().mockImplementation(async () => Promise.resolve(true));
+
+            await flushPromises();
+            await connector.subscribe({ kind: 'foo', namespace: 'bar' }, handler);
+
+            await flushPromises();
+            expect(handler).toHaveBeenCalledTimes(1);
+            expect(handler).toHaveBeenCalledWith('foo');
+            expect(mockChannel.ack).toHaveBeenCalledTimes(1);
+            expect(mockChannel.ack).toHaveBeenCalledWith({
+                content: { toString: () => '{ "event": "foo" }' },
+            });
+        });
+
+        it('it should call the subscription handler and unacknowledge if not ok', async () => {
+            const mockChannel = makeChannel({
+                assertQueue: jest.fn().mockImplementation(() => Promise.resolve()),
+                consume: (___, callback) => {
+                    callback({
+                        content: { toString: () => '{ "event": "foo" }' },
+                    });
+                },
+            });
+            const mockConnection = makeConnection(mockChannel);
+
+            // tslint:disable-next-line: no-any
+            (connect as any).mockImplementation(async (): Promise<Connection> => mockConnection);
+            const connector = new AMQPConnector<{}>(url, channel(), [], [], 'service');
+            const handler = jest.fn().mockImplementation(async () => Promise.resolve());
+
+            await flushPromises();
+            await connector.subscribe({ kind: 'foo', namespace: 'bar' }, handler);
+
+            await flushPromises();
+            expect(handler).toHaveBeenCalledTimes(1);
+            expect(logger.warn).toHaveBeenCalledTimes(1);
+            expect(logger.warn).toHaveBeenCalledWith('eventHandlerFailure');
```

Comment:
Worth keeping in mind for the future: standardising on log message format will make things a _*heck*_ of a lot easier later on when you need to trace across a few services.